Source/Scene/Instanced3DModel3DTileContentProvider.js

```diff
@@ -0,0 +1,233 @@
+        return array.subarray(offset, offset + length);
+    }
+
+    /**
+     * DOC_TBA
+     *
+     * Use Cesium3DTile#requestContent
+     */
+    Instanced3DModel3DTileContentProvider.prototype.request = function() {
+        var that = this;
+
+        this.state = Cesium3DTileContentState.LOADING;
+
+        function failRequest(error) {
+            that.state = Cesium3DTileContentState.FAILED;
+            that.readyPromise.reject(error);
+        }
+
+        loadArrayBuffer(this._url).then(function(arrayBuffer) {
+            var uint8Array = new Uint8Array(arrayBuffer);
+            var magic = getStringFromTypedArray(getSubarray(uint8Array, 0, Math.min(4, uint8Array.byteLength)));
+            if (magic !== 'i3dm') {
+                throw new DeveloperError('Invalid Instanced 3D Model. Expected magic=i3dm. Read magic=' + magic);
+            }
+
+            var view = new DataView(arrayBuffer);
+            var byteOffset = 0;
+
+            byteOffset += sizeOfUint32;  // Skip magic number
+
+            //>>includeStart('debug', pragmas.debug);
+            var version = view.getUint32(byteOffset, true);
+            if (version !== 1) {
+                throw new DeveloperError('Only Instanced 3D Model version 1 is supported. Version ' + version + ' is not.');
+            }
+            //>>includeEnd('debug');
+            byteOffset += sizeOfUint32;
+
+            var batchTableLength = view.getUint32(byteOffset, true);
+            byteOffset += sizeOfUint32;
+
+            var gltfLength = view.getUint32(byteOffset, true);
+            byteOffset += sizeOfUint32;
+
+            var gltfFormat = view.getUint32(byteOffset, true);
+            byteOffset += sizeOfUint32;
+
+            var instancesLength = view.getUint32(byteOffset, true);
+            byteOffset += sizeOfUint32;
+
+            var batchData = new Cesium3DTileBatchData(that, instancesLength);
+            that._batchData = batchData;
+            if (batchTableLength > 0) {
+                var batchTableString = getStringFromTypedArray(getSubarray(uint8Array, byteOffset, batchTableLength));
+                batchData._batchTable = JSON.parse(batchTableString);
+                byteOffset += batchTableLength;
+            }
+
+            var gltfView = new Uint8Array(arrayBuffer, byteOffset, gltfLength);
+            byteOffset += gltfLength;
+
+            var instancesDataSizeInBytes = instancesLength * instanceSizeInBytes;
+            var instancesView = new DataView(arrayBuffer, byteOffset, instancesDataSizeInBytes);
+            byteOffset += instancesDataSizeInBytes;
+
+            // TODO : what is the model's cache key if it doesn't have a url? Url of the tile?
+            // Create model instance collection
+            var collectionOptions = {
+                instances : [],
+                batchData : batchData,
+                boundingVolume : undefined, // TODO : what is the best way to get the tile's bounding volume?
```

Comment:
See the content header from the 3D tileset.

For example: https://github.com/AnalyticalGraphicsInc/cesium/blob/3d-tiles/Source/Scene/Points3DTileContentProvider.js#L54